app/src/server/tasks/fineTuning/deployFineTune.task.ts

```diff
@@ -0,0 +1,83 @@
+import fs from "fs/promises";
+import path from "path";
+import os from "os";
+import { env } from "~/env.mjs";
+import { prisma } from "~/server/db";
+import defineTask from "../defineTask";
+import url from "url";
+import dedent from "dedent";
+import { $ } from "execa";
+
+export type DeployFineTuneJob = {
+  fineTuneId: string;
+};
+
+export const deployFineTuneTask = defineTask<DeployFineTuneJob>("deployFineTune", async (task) => {
+  await deployFineTune(task.fineTuneId);
+});
+
+const dirname = path.dirname(url.fileURLToPath(import.meta.url));
+export const trainerDirectory = path.join(dirname, "..", "..", "..", "..", "..", "trainer");
+
+export async function deployFineTune(fineTuneId: string) {
+  const fineTune = await prisma.fineTune.findUnique({
+    where: { id: fineTuneId },
+  });
+
+  if (!fineTune) {
+    throw new Error("Trying to deploy a model that does not exist");
+  }
+
+  if (!fineTune.huggingFaceModelId) {
+    throw new Error("Trying to deploy a model that does not have a Hugging Face model ID");
+  }
+
+  await prisma.fineTune.update({
+    where: { id: fineTuneId },
+    data: {
+      status: "DEPLOYING",
+      deploymentStartedAt: new Date(),
+    },
+  });
+
+  const modalDeployId = `ft-${env.NODE_ENV}-${fineTuneId}`;
+
+  // Inline setupTemporaryEnvironment
+  const tempDir = await fs.mkdtemp(path.join(os.tmpdir(), "deploy-"));
```

Comment:
Should we incorporate the deploy id into the tempDir name?

Reply:
Now outdated